ambry-clustermap/src/main/java/com.github.ambry.clustermap/FixedBackoffResourceStatePolicy.java

```diff
@@ -55,6 +55,7 @@ public void onError() {
       if (count == failureCountThreshold) {
         logger.error("Resource " + resource + " has gone down");
       }
+      logger.trace("Resource " + resource + " remains in down state at time " + SystemTime.getInstance().milliseconds() + "; Adding downtime of " + retryBackoffMs + " ms");
```

Comment:
when trace logging, do not concatenate. Trace logging is not printed most of the times but you suffer string concatenation cost anyway.

logger provides a nice way to avoid this through formats. The strings will be concatenated only if they are going to be printed. This statement should be written as
```
logger.trace("Resource {} remains in down state at time {}; Adding downtime of {} ms", resource, SystemTime.getInstance().milliseconds(), retryBackoffMs);
```